samcli/lib/test_runner/generate_env_vars.py

```diff
@@ -0,0 +1,94 @@
+import yaml
+from typing import List
+from samcli.commands.exceptions import FailedArnParseException
+
+
+class FargateRunnerArnMapGenerator:
+    def _get_env_var_name(self, resource_arn: str) -> str:
+        """
+        Generates a valid environment variable name from a given resource ARN.
+
+        Environment variable names will be the resource name with an underscore prepended, all dashes replaced by underscores, and in all upper case.
+
+        e.g. `arn:aws:lambda:us-east-1:123456789012:function:my-lambda-function` => `_MY_LAMBDA_FUNCTION`
+
+        Parameters
+        ----------
+        resource_arn : str
+            The string of the resource arn from which the resource name is extracted and the environment variable name is created.
+
+        Returns
+        -------
+        str
+            An environment variable name representative of the resource name.
+
+        Raises
+        ------
+        FailedArnParseException
+            If a resource name/id could not be extracted from the resource_arn.
+
+            NOTE: https://docs.aws.amazon.com/quicksight/latest/APIReference/qs-arn-format.html
+        """
+
+        # Extract resource-id from ARN
+        # Cases:
+        # 1) arn:partition:service:region:account-id:resource-id
+        # 2) arn:partition:service:region:account-id:resource-type/resource-id
+        # 3) arn:partition:service:region:account-id:resource-type:resource-id
+        components = resource_arn.split(":")
```

Comment:
Do we have any existing methods to parse ARNs? If not, this is a prime candidate for a separate function.

Reply:
Unfortunately we do not, so yes I'll abstract out the parsing code.